requirements.txt

```diff
@@ -1,7 +1,7 @@
 # Requirements for building PEPs with Sphinx
 Pygments >= 2.9.0
 Sphinx >= 4.0.2
-docutils >= 0.17.1
+docutils >= 0.19.0
```

Comment:
This seems rather aggressive, especially just for the RSS feed generation rather than the actual Sphinx build, which everything except the CIs is never going to run. Docutils 0.19 was only released less than a month ago, and Sphinx 5.1.0, the first version supporting it, was released only 10 days ago—which makes the 4.0.2 constraint above rather misleading and requires Sphinx 5.x for the first time, when itself is still only a couple month or two old. Can we revert this change and rely on the fact that the CIs, the only thing running the RSS gen aside from ourselves for testing, will always grab the latest version?

```suggestion
docutils >= 0.17.1
```

Reply:
Can do, and if this were a library I can see the benefits.

But the PEPs website is a standalone application so we don't need to support a wide range of versions, and should rather keep it narrower to avoid running different things in production and local/test.

(And applications should really pin dependencies and regularly bump them.)

We also have tooling to use venvs to keep things isolated to this app.

Is there a specific problem using recent versions of Docutils/Sphinx?

Reply:
Sorry, my comment didn't actually post before merging.

-----

I agree with Hugo on requirements--the Sphinx constraint is technically wrong but reflects the features we require from core Sphinx--rendering the PEPs locally with Docutils 0.17 will still work at the moment, we aren't gating within the rendering process itself.

A

Reply:
> Is there a specific problem using recent versions of Docutils/Sphinx?

Not really, no, since this isn't a Python package and I can just keep installed whatever versions I want locally, and on CI we get the latest anyway, so it doesn't mean as much as I thought. I'm just naturally used to working on library and scientific code where trying to ensure compatibility and minimize failed solves is a constant concern, as well as the fact that now the PEPs require Sphinx 5, the Python Docs require Sphinx 4 and some of the docs sites I maintain still require Sphinx 3 due to being stuck on an old theme and plugins that were never updated for Sphinx >=4 (I really need to do something about that...)